rust/arrow/src/datatypes.rs

```diff
@@ -920,6 +930,32 @@ make_signed_numeric_type!(Int64Type, i64x8);
 make_signed_numeric_type!(Float32Type, f32x16);
 make_signed_numeric_type!(Float64Type, f64x8);
 
+#[cfg(simd)]
+pub trait ArrowFloatNumericType: ArrowNumericType {
+    fn pow(base: Self::Simd, raise: Self::Simd) -> Self::Simd;
```

Comment:
We could avoid creating this trait by using `num::Float` as a constraint for floaing types. See #9313, though I wasn't doing it for a SIMD implementation

Reply:
For non-SIMD this would work indeed. However I could not proof my `SIMD` type had the `powf` method without this trait/ implementation. Or am I missing something?

Reply:
You're right, if it's not possible to only add the trait for SIMD types, then it's a reasonable change to make.
The downside's that the trait's going to grow when we add more compute kernels (with SIMD support).